test/blackbox-tests/test-cases/private-modules/run.t

```diff
@@ -0,0 +1,45 @@
+  $ dune build --root accessible-via-public
+  Entering directory 'accessible-via-public'
+        runfoo alias default
+  private module bar
+
+  $ dune build --root inaccessible-in-deps 2>&1 | grep -v "cd _build"
+  Entering directory 'inaccessible-in-deps'
+        ocamlc .foo.eobjs/foo.{cmi,cmo,cmt} (exit 2)
+  File "foo.ml", line 1, characters 0-5:
+  Error: Unbound module X
+
+  $ dune build --root excluded-from-install-file
+  Entering directory 'excluded-from-install-file'
+  lib: [
+    "_build/install/default/lib/lib/META" {"META"}
+    "_build/install/default/lib/lib/foo.ml" {"foo.ml"}
+    "_build/install/default/lib/lib/foo/bar.cmi" {"foo/bar.cmi"}
+    "_build/install/default/lib/lib/foo/bar.cmt" {"foo/bar.cmt"}
+    "_build/install/default/lib/lib/foo/bar.cmx" {"foo/bar.cmx"}
+    "_build/install/default/lib/lib/foo/bar.ml" {"foo/bar.ml"}
+    "_build/install/default/lib/lib/foo/lib.foo.dune" {"foo/lib.foo.dune"}
+    "_build/install/default/lib/lib/foo/lib_foo$ext_lib" {"foo/lib_foo$ext_lib"}
+    "_build/install/default/lib/lib/foo/lib_foo.cma" {"foo/lib_foo.cma"}
+    "_build/install/default/lib/lib/foo/lib_foo.cmxa" {"foo/lib_foo.cmxa"}
+    "_build/install/default/lib/lib/foo/lib_foo.cmxs" {"foo/lib_foo.cmxs"}
+    "_build/install/default/lib/lib/foo/priv2.cmt" {"foo/priv2.cmt"}
+    "_build/install/default/lib/lib/foo/priv2.cmx" {"foo/priv2.cmx"}
+    "_build/install/default/lib/lib/foo/priv2.ml" {"foo/priv2.ml"}
```

Comment:
Do we need to install any more that the .cmx for private modules?

Reply:
Isn't it still necessary for cross module optimization?

Reply:
Only the .cmx is necessary, the rest is not.

Reply:
Oh, you mean the cmt and the .ml files? I think they should be installed. Users might be reading the .ml file of a public module and want to jump to a private module they see. The .cmt file would then be useful too.

Reply:
Ok